ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/EthSigner.java

```diff
@@ -92,6 +96,41 @@ public void run() {
     runner.start();
   }
 
+  private HttpClientOptions applyConfigTlsSettingsTo(final HttpClientOptions input) {
+    final HttpClientOptions result = new HttpClientOptions(input);
+    boolean tlsIsRequired =
+        config.getWeb3TrustStoreOptions().isPresent()
+            || config.getClientCertificateOptions().isPresent();
+
+    if (tlsIsRequired) {
+      result.setSsl(true);
+      try {
+        if (config.getWeb3TrustStoreOptions().isPresent()) {
+          result.setPfxTrustOptions(convertFrom(config.getWeb3TrustStoreOptions().get()));
+        } else {
+          result.setTrustAll(true);
+        }
+      } catch (final IOException e) {
+        throw new InitializationException("Failed to load web3 trust store.", e);
+      }
+
+      if (config.getClientCertificateOptions().isPresent()) {
+        try {
+          result.setPfxTrustOptions(convertFrom(config.getClientCertificateOptions().get()));
```

Comment:
can `getWeb3TrustStoreOptions` and `getClientCertificateOptions` both present at same time? If they can, the line above just overwrote trust options on result.

Reply:
yeah - that should have been KeyCert....:smh: